### ISSUE TITLE
[@astrojs/image] local file path not working upon build

### ISSUE THREAD

**Issue by lubomirblazekcz** [withastro/astro#5008]
### What version of `astro` are you using?

1.4.5

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

https://docs.astro.build/en/guides/integrations-guide/image/#src

Here in docs it's mentioned following

> For images located in your project’s src: use the file path relative to the src directory. (e.g. src="../assets/source-pic.png")

With this usage the images are not proccessed to dist directory, the path is proccessed correctly in `html` files, but image files are not coppied to `assets` directory.

### Link to Minimal Reproducible Example

https://github.com/lubomirblazekcz/astro-image-bug-repro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by that-joao** [comment]
As per the [docs](https://docs.astro.build/en/guides/integrations-guide/image/#local-images), I think the issue here is that you are not importing the image. Taking your reproducible example, Instead of the following:

```
<Image src="/src/assets/image.jpg" alt="" format="webp" width="200" height="300" loading="lazy" />
```

You should try like this:

```
<Image src="{import('../assets/image.jpg')}" alt="" format="webp" width="200" height="300" loading="lazy" />
```

**Comment by lubomirblazekcz** [comment]
Yes it works with import, but I would expect it to work without import too. But if this is the intended usage then these parts in docs should mention it - https://docs.astro.build/en/guides/integrations-guide/image/#src

**Comment by matthewp** [comment]
@tony-sull does the Image component do something special to know that the `src` is coming from an import and not a string literal?

**Comment by tony-sull** [comment]
The local image files always need to be imported, either in the front matter or directly in line like `src={import("./hero.jpg")}`

The [examples section](https://www.npmjs.com/package/@astrojs/image#examples) of the Readme shows that but we probably should call that out in the `<Image />` section as well to make that more clear!

**Comment by wulinsheng123** [comment]
mark

**Comment by stephanbogner** [comment]
**How hard would this be to implement?**
(To make it work with a string literal)

So far **I was using astro-imagetools** mainly because of 2 advantages/features:
- Works with Markdown files (where using `import` is quite clunky / not user friendly)
- Doesn't require any width/height specification (which if I understand should only apply to remote images so maybe not even an issue)
... unfortunately it's quite broken in newer Astro versions which is why I am looking for alternatives.

I could potentially contribute, but if it's a hard problem I'd look into [JamPack](https://github.com/divriots/jampack) as an alternative approach.

PS: I have now done a few client website using Astro and dealing with image optimizations (mainly within Markdown files) is probably the biggest hackaround I still have.

**Comment by Princesseuh** [comment]
Hello! We're heading into maintenance mode on `@astrojs/image` and won't be adding new features. The new `astro:assets` feature make this situation a bit clearer and hopefully addresses some of the issues presented in the comments here. (Namely, assets in Markdown).
